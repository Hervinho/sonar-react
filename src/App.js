import { useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Modal, Button} from 'react-bootstrap';

const App = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <>
      <Button type="primary" onClick={() => setIsOpen(true)} style={{ backgroundColor: '#0054a6'}}>
        Cliquez Moi
      </Button>

      <Modal show={isOpen} backdrop={'static'}>
        <Modal.Body>
          Bonjour Laval!
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={() => setIsOpen(false)}>
            Fermez
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};

export default App;
