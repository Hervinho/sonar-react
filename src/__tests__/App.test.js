import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import App from '../App';

test('should render page', () => {
  render(<App />);
  const openBtn = screen.getByRole("button", { name: "Cliquez Moi" });
  const modal = screen.queryByRole("dialog");
  expect(openBtn).toBeInTheDocument();
  expect(openBtn).toBeVisible();
  expect(modal).not.toBeInTheDocument();
});

test('should open modal', () => {
  render(<App />);
  const openBtn = screen.getByRole("button", { name: "Cliquez Moi" });
  fireEvent.click(openBtn);
  const modal = screen.getByRole("dialog");

  // assert
  expect(screen.getByText('Bonjour Laval!')).toBeVisible();
  expect(screen.getByRole("button", { name: "Fermez" })).toBeVisible();
  expect(modal).toBeVisible();
  expect(modal).toBeInTheDocument();
  expect(modal).toHaveClass('show');
});

test('should close modal', () => {
  render(<App />);
  const openBtn = screen.getByRole("button", { name: "Cliquez Moi" });
  fireEvent.click(openBtn);

  const modal = screen.queryByRole("dialog");
  const closeBtn = screen.getByRole("button", { name: "Fermez" });
  fireEvent.click(closeBtn);

  setTimeout(() => {
    expect(openBtn).toBeVisible();
    expect(screen.queryByText('Bonjour Laval!')).toBeNull();
    expect(screen.queryByText('Bonjour Laval!')).not.toBeVisible();
    expect(closeBtn).toBeNull();
    expect(modal).not.toBeInTheDocument();
  }, 500);
});

test('main button should have the correct background color', () => {
  render(<App/>);
  const openBtn = screen.getByRole("button", { name: "Cliquez Moi" });
  const computedStyle = window.getComputedStyle(openBtn);
  expect(computedStyle.backgroundColor).toBe('rgb(0, 84, 166)');
});

test('modal should not close when user clicks outside', () => {
  render( < App/> );
  const openBtn = screen.getByRole("button", { name: "Cliquez Moi" });
  fireEvent.click(openBtn);
  expect(screen.getByRole('dialog')).toBeInTheDocument();
  userEvent.click(screen.getByRole('dialog'));
  expect(screen.getByRole('dialog')).toBeInTheDocument();
});
