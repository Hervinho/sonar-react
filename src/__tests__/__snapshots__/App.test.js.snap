// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can open modal 1`] = `
<body
  class="modal-open"
  data-rr-ui-modal-open=""
  style="padding-right: 1024px;"
>
  <div>
    <button
      class="btn btn-primary"
      style="background-color: rgb(0, 84, 166);"
      type="primary"
    >
      Cliquez Moi
    </button>
  </div>
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal show"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-body"
        >
          Bonjour Laval!
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            type="button"
          >
            Fermez
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders page 1`] = `
<body>
  <div>
    <button
      class="btn btn-primary"
      style="background-color: rgb(0, 84, 166);"
      type="primary"
    >
      Cliquez Moi
    </button>
  </div>
</body>
`;
